scripts/rebuild_proxies.py

```diff
@@ -0,0 +1,100 @@
+# Read the output from 'multipass list --output json' and parse it to an object
+# that can be used to rebuild the proxies in the traefik and haproxy config files.
+
+import json
+import os
+import sys
+
+# Read the output from running 'multipass list --output json' command and parse it to an object
+def read_multipass_list():
+    multipass_list = os.popen('multipass list --format json').read()
+    multipass_list = json.loads(multipass_list)
+    return multipass_list
+
+# Go through the multipass list and build a list of the proxies
+
+def build_proxy_list(multipass_list):
+    proxy_list = []
+    for instance in multipass_list['list']:
+        instance_hostname = instance['name']
+        if instance_hostname and instance['ipv4']:
+            instance_ip = instance['ipv4'][0]
+            proxy_list.append({
+                'name': instance_hostname,
+                'ip': instance_ip,
+                # 'port': instance_ip.split('.')[3]
+            })
+
+    return proxy_list
+
+# Write the proxy list to the files
+
+def write_proxy_list(proxy_list):
+    traefik_config_file = open('./edgebox.toml', 'w')
+    traefik_generated_config = ""
+    # haproxy_config_file = open('/etc/haproxy/haproxy.cfg', 'w')
+    # haproxy_generated_config = ""
+    server_count = 0
+    for proxy in proxy_list:
+
+        server_count += 1
+
+        # Prepare traefik entries
+        # They should look like this:
+        # [http.routers.adm-edgebox-wildcard]
+        #     rule = "HostRegexp(`adm-{subdomain:[a-z0-9]+}.edgebox.io`)"
+        #     service = "adm-edgebox-service"
+        #     priority = 1
+
+        # [http.routers.adm-edgebox]
+        #     rule = "Host(`adm.edgebox.io`)"
+        #     service = "adm-edgebox-service"
+        #     priority = 2
+
+        # [http.services.adm-edgebox-service.loadBalancer]
+        #     [[http.services.adm-edgebox-service.loadBalancer.servers]]
+        #         url = "http://10.139.121.36:80"
+
+        traefik_generated_config += "[http.routers." + proxy['name'] + "-edgebox-wildcard]\n"
```

Comment:
Perhaps we could improve this with leveraging the toml package and writing in a nicer way.

See: https://www.geeksforgeeks.org/read-and-write-toml-files-using-python/

Reply:
Absolutely 🙌 looks way cleaner, thanks for the suggestion!